app/javascript/controllers/challenge_detail_controller.js

```diff
@@ -0,0 +1,94 @@
+import { Controller } from "@hotwired/stimulus"
+
+export default class extends Controller {
+  static targets = ["copyButton", "modal", "button", "content", "menu", "section"]
+
+  connect() {
+    window.addEventListener('hashchange', () => this.showActiveSection())
+  }
+
+  // Navbar
+  showActiveSection() {
+    document.querySelectorAll('.challenge-section').forEach(section => {
+      section.style.display = 'none'
+    })
+
+    document.querySelectorAll('.nav.navbar-nav li').forEach(item => {
+      item.classList.remove('active')
+    })
+
+    // Get the hash from URL or default to overview
+    let hash = window.location.hash || '#overview'
+    hash = hash.replace('#', '')
+
+    // Show the selected section
+    const activeSection = document.getElementById(hash)
+    if (activeSection) {
+      activeSection.style.display = 'block'
+      
+      activeSection.style.display = 'none'
+      activeSection.offsetHeight
+      activeSection.style.display = 'block'
+    }
+
+    // Add active class to current nav item
+    const activeNavItem = document.querySelector(`.nav.navbar-nav li a[href$="#${hash}"]`)?.parentElement
+    if (activeNavItem) {
+      activeNavItem.classList.add('active')
+    }
+  }
+
+  // Share social icons
+  toggle(event) {
+    event.preventDefault()
+    this.menuTarget.classList.toggle("is-visible")
+  }
+
+  share(event) {
+    event.preventDefault()
+    const type = event.currentTarget.dataset.shareType
+  
+    switch(type) {
+      case 'facebook':
+        window.open('https://www.facebook.com/ChallengeGov', '_blank')
+        break
+      case 'twitter':
+        window.open('https://www.twitter.com/ChallengeGov', '_blank')
+        break
+      case 'linkedin':
+        window.open('https://www.linkedin.com/company/challengegov/', '_blank')
+        break
```

Comment:
These links work a bit differently in React, they compose a post on the different social sites. Let's try to research how to generate these kinds of social post links dynamically with things like the URL to the challenge. apparently React handles it automatically with a module dependency, but we don't have to copy that behavior 100%.